src/Compilers/Core/Portable/Symbols/ITypeSymbolInternal.cs

```diff
@@ -38,5 +38,11 @@ internal interface ITypeSymbolInternal : INamespaceOrTypeSymbolInternal
         /// This API and <see cref="ISymbolInternal.GetISymbol"/> should return the same object.
         /// </summary>
         ITypeSymbol GetITypeSymbol();
+
+        /// <summary>
+        /// For SZArrays, returns the type of elements stored in the array.
+        /// Returns null on all other kinds of types.
+        /// </summary>
+        ITypeSymbolInternal GetElementTypeIfSZArray();
```

Comment:
>ITypeSymbolInternal GetElementTypeIfSZArray(); [](http://example.com/codeflow?start=8&length=46)

I think that instead we should add ```IArrayTypeSymbolInternal``` with all the necessary APIs complete the task (pretty much simply exposing necessary APIs that we already have on the internal array symbol).